### PR TITLE
Extend flexibility of cluster parameters

### PR DESCRIPTION
**Pull request by imiller0**
Convert nmstate to template with basic networking parameters supplied in
inventory file. Allow flexibility in providing cluster networking info
(allow defaults for some networking). Additional documentation in
inventory file.

Signed-off-by: Ian Miller <imiller@redhat.com>

### CODE REVIEW

**Review by yrobla — CHANGES_REQUESTED** [review]
So i think it's a good idea, but i would also let the freedom of writing your own nmstate file. That's because what is shown there, is just a sample, but nmstate can have multiple formats. You can have ipv6 config (in another pr), you may want to add vlan/bonding... so many cases to just reflect on a template.
Can we find some solution to generate a simple template, but also leave freedom to configure whatever is needed?

**Comment by serngawy** [comment]
Just to clarify Yalonda comment. If I have my nnmstate file and set the nmstate_path= to it and ignore other parameters. Does it will just pickup my nmstate file ?

File: create-cluster-sno/roles/chrony_day0/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+- name: Create the workload-partitioning MachineConfig manifest
```
Comment (lack) [inline create-cluster-sno/roles/chrony_day0/tasks/main.yml:2]: This name doesn't match the action ;)

**Comment by imiller0** [comment]
Yes, the nmstate_path variable controls what file is pulled in. The main change I made was to treat it as a template so Ansible will do the substitutions. I think this allows the flexibility @yrobla is looking for along with some scalability by allowing the values to come from the inventory.

File: create-cluster-sno/roles/chrony_day0/tasks/main.yml (hunk 1 above)
Comment (imiller0) [reply create-cluster-sno/roles/chrony_day0/tasks/main.yml]: Fixed :)

**Review by serngawy — APPROVED** [review]

**Review by yrobla — APPROVED** [review]

**Pull request merged**